### PR TITLE
Removes immutable property assertion

### PR DESCRIPTION
**Pull request by pzuraq**
Based on my reading in [this issue on private class fields](https://github.com/tc39/proposal-class-fields/issues/106), specifically [this comment](https://github.com/tc39/proposal-class-fields/issues/106#issuecomment-421556491) which sums up the most recent state of discussion, it seems that native proxies will _not_ be able to interoperate with private class fields, and will hard error if they exist. 

This represents a fundamental tradeoff: security vs. expressiveness. TC39 seems to have intended Proxies to be used primarily for security, enabling users to provide membranes around untrusted code. Private fields cannot bypass this security layer without violating that security, and they can't expose themselves to it without violating the "hard private" requirement they've given themselves (which they do not seem to be budging on).

In other words, using this pattern seems like it _could_ make future Javascript features difficult/impossible to use, and I'm wary of that. Despite the assertion's potential usefulness, this seems like it would be a very bad place to be if/when private fields become standard.

I think for us to be confident with using a recursive proxy technique, TC39 would have to either establish that Proxies _will_ always be usable for expressive purposes like ours, either directly or via a variation such as the [`Proxy.transparent` proposal](https://github.com/littledan/proposal-proxy-transparent). Until then, we have to assume that they could break anything in favor of security.

### CODE REVIEW

**Comment by pzuraq** [comment]
One possibility that just occurred to me is that we could continue to use the private fields babel transform in development mode even after private fields are moved to stage 4. This would allow us to keep the immutability assertion, though it could be risky (exposing internals during development is potentially very bad).
